File: androidApp/src/main/kotlin/website/lolegrand/personapp/detail/adapter/DetailAdapter.kt
package website.lolegrand.personapp.detail.adapter

import website.lolegrand.personapp.detail.model.DetailGenderUI
import website.lolegrand.personapp.detail.model.DetailPersonUI
import website.lolegrand.personapp.local.model.GenderEntity
import website.lolegrand.personapp.local.model.PersonEntity

fun PersonEntity.toDetailPersonUI(): DetailPersonUI =
    DetailPersonUI(
        id = id,
        name = name,
        address = address,
        longitude = longitude,
        latitude = latitude,
        email = email,
        age = age,
        phone = phone,
        cell = cell,
        pictureUrl = pictureUrl,
        gender = gender.toDetailGenderUI(),
    )

fun GenderEntity.toDetailGenderUI(): DetailGenderUI =
    when (this) {
        GenderEntity.MALE -> DetailGenderUI.MALE
        GenderEntity.FEMALE -> DetailGenderUI.FEMALE
    }

File: iosApp/iosApp/Detail/Adapter/DetailAdapter.swift
//
//  DetailAdapter.swift
//  iosApp
//
//  Created by Louis Legrand on 03/02/2025.
//  Copyright © 2025 orgName. All rights reserved.
//

import Foundation
import SwiftUI
import Shared

extension GenderEntity {
    
    func toDetailGenderUI() -> DetailGenderUI {
        switch self {
        case .male: return DetailGenderUI.Male
        case .female: return DetailGenderUI.Female
        }
    }
    
}

extension PersonEntity {
    
    func toDetailPersonUI() -> DetailPersonUI {
        return DetailPersonUI(gender: gender.toDetailGenderUI(), name: name, address: address, longitude: longitude, latitude: latitude, email: email, age: age, id: id, phone: phone, cell: cell, pictureUrl: pictureUrl)
    }
    
}

